### PR TITLE
Add Azure sign-in

### PR DESCRIPTION
**Pull request by dimaryaz**

### CODE REVIEW

File: catalog/app/containers/Auth/messages.js [hunk 1]
```diff
@@ -79,6 +79,24 @@ export default defineMessages({
     defaultMessage:
       'No Quilt user linked to this OneLogin account. Notify your Quilt administrator.',
   },
+  ssoAzureUse: {
+    id: `${scope}.SSO.Azure.use`,
+    defaultMessage: 'Sign in with Microsoft',
```
Comment (dimaryaz) [inline catalog/app/containers/Auth/messages.js:84]: Note that [Microsoft's branding guidelines](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-add-branding-in-azure-ad-apps) state that in the end-user UI, it should be called "Microsoft", not "Azure".

File: catalog/app/containers/Auth/microsoft-logo.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="21" height="21" viewBox="0 0 21 21"><title>MS-SymbolLockup</title><rect x="1" y="1" width="9" height="9" fill="#f25022"/><rect x="1" y="11" width="9" height="9" fill="#00a4ef"/><rect x="11" y="1" width="9" height="9" fill="#7fba00"/><rect x="11" y="11" width="9" height="9" fill="#ffb900"/></svg>
```
Comment (nl0) [inline catalog/app/containers/Auth/microsoft-logo.svg:1]: @dimaryaz pls run `npx svgo --multipass  -i app/containers/Auth/microsoft-logo.svg`

File: catalog/app/containers/Auth/SSOAzure.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+const MUTEX_POPUP = 'sso:azure:popup'
+const MUTEX_REQUEST = 'sso:azure:request'
+
+export default function SSOAzure({ mutex, next, ...props }) {
+  const cfg = Config.useConfig()
+  invariant(!!cfg.azureClientId, 'Auth.SSO.Azure: config missing "azureClientId"')
+  invariant(!!cfg.azureTenantId, 'Auth.SSO.Azure: config missing "azureTenantId"')
+  const authenticate = Azure.use({
+    clientId: cfg.azureClientId,
+    authUrl: `https://login.microsoftonline.com/${cfg.azureTenantId}/oauth2/v2.0/authorize`,
```
Comment (nl0) [inline catalog/app/containers/Auth/SSOAzure.js:31]: i think we should pass `tenantId` as a parameter and construct the url inside `Azure.use`

**Review by nl0 — APPROVED** [review]
lgtm, some cosmetic change requests

**Review by akarve — APPROVED** [review]

**Pull request merged**